sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
I can see how this review works on its own, but not sure how it fits into the context of DX-1645 - if API can/will toggle scheduler debug mode, why have it in config?  Just more flexibility to bring it up in debug mode from the start?